cpp/src/arrow/compute/kernels/vector_sort.cc

```diff
@@ -137,6 +138,30 @@ struct ChunkedArrayResolver {
   mutable int64_t cached_chunk_;
 };
 
+// We could try to reproduce the concrete Array classes' facilities
+// (such as cached raw values pointer) in a separate hierarchy of
+// physical accessors, but doing so ends up too cumbersome.
+// Instead, we simply create the desired concrete Array objects.
+ArrayVector GetPhysicalChunks(const ChunkedArray& chunked_array,
```

Comment:
Doesn't this duplicate `ChunkedArray::View`?

Reply:
This is much simpler and doesn't get through the sophisticated `View()` machinery.